playwright/e2eTests/api-delegering.spec.ts

```diff
@@ -1,100 +1,228 @@
-      'INTERESSANT KOMPATIBEL TIGER ASOrg.nr. 310661414',
-    );
+    await apiDelegations.delegateAPI(standardApiDetails.name, standardOrgUser.orgNumber);
+    await apiDelegations.verifyConfirmationPage(standardApiDetails, standardOrgUser);
 
     //Delegate API to same Org to which API was delegated before
-    await apiDelegations.chooseApiToDelegate(
-      'Automation Regression',
-      'INTERESSANT KOMPATIBEL TIGER ASOrg.nr. 310661414',
-    );
+    await page.goto(process.env.BASE_URL + '/ui/profile');
+    await apiDelegations.delegateAPI(standardApiDetails.name, standardOrgUser.orgNumber);
+    await apiDelegations.verifyConfirmationPage(standardApiDetails, standardOrgUser);
   });
 
   test('Verify filtering of API providers in API delagation and verify Forrige/ Neste buttons', async ({
     login,
-    logoutUser,
-    context,
     apiDelegations,
-    page,
   }) => {
-    await login.gotoLoginPage('14824497789', page);
-    await login.chooseReportee('AKTVERDIG RETORISK APE', page);
+    await login.loginWithUser(standardDaglUser.id);
+    await login.chooseReportee(standardDaglUser.reportee);
 
     //delete delegated API
-    await apiDelegations.deleteDelegatedAPI();
+    await apiDelegations.deleteDelegatedAPIs();
 
     //API-delegations
-    await apiDelegations.apiFiltering();
-    await apiDelegations.delegatedAPIOverviewPage(
-      'Maskinporten Schema - AM - K6Testdepartement',
-      'INTERESSANT KOMPATIBEL TIGER ASOrg.nr. 310661414',
+    await apiDelegations.delegateAPI(standardApiDetails.name, standardOrgUser.orgNumber);
+    await apiDelegations.verifyConfirmationPage(
+      { name: standardApiDetails.name, department: standardApiDetails.department },
+      { orgNumber: standardOrgUser.orgNumber, reportee: standardOrgUser.reportee },
     );
   });
+
+  test('Delegate api to an organization by selecting from API providers filter', async ({
+    login,
+    apiDelegations,
+    page,
+  }) => {
+    //Login and cleanup state before running test
+    await login.loginWithUser(standardDaglUser.id);
+    await login.chooseReportee(standardDaglUser.reportee);
+    await apiDelegations.deleteDelegatedAPIs();
+
+    //Replace this pls?
+    await apiDelegations.delegateAPI(standardApiDetails.name, standardOrgUser.orgNumber);
+
+    await expect(apiDelegations.giveAccessToNewApiHeading).toBeVisible();
+
+    await expect(page.getByText(standardApiDetails.name)).toBeVisible();
+    await expect(page.getByText(standardApiDetails.department)).toBeVisible();
+    await expect(page.getByText(`Org.nr. ${standardOrgUser.orgNumber}`)).toBeVisible();
+
+    await page.getByRole('button', { name: 'Bekreft' }).click();
+
+    await expect(
+      page.getByRole('heading', { name: 'Disse api-delegeringene ble gitt' }),
+    ).toBeVisible();
+
+    await expect(page.getByText(standardOrgUser.reportee)).toBeVisible();
+    await expect(page.getByText(standardApiDetails.name)).toBeVisible();
+  });
+});
```

Comment:
Maybe we can have all tests in same test script here instead of adding them seperately here?

Reply:
Probably a good idea? Haven't put much thought into how to organizing the test / describe steps, really, so should probably update that